verification/curator-service/ui/src/components/LinelistTable.tsx

```diff
@@ -632,13 +658,60 @@ class LinelistTable extends React.Component<Props, LinelistTableState> {
                     actions={
                         this.props.user.roles.includes('curator')
                             ? [
+                                  {
+                                      icon: (): JSX.Element => (
+                                          <Button variant="outlined">
+                                              {this.state.totalNumRows ===
+                                              this.state.numSelectedRows
+                                                  ? 'Unselect'
+                                                  : 'Select'}{' '}
+                                              all {this.state.totalNumRows} rows
+                                          </Button>
+                                      ),
+                                      tooltip: `
+                                      ${
+                                          this.state.totalNumRows ===
+                                          this.state.numSelectedRows
+                                              ? 'Unselect'
+                                              : 'Select'
+                                      } all rows across pages`,
+                                      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                                      onClick: async (
+                                          _: any,
+                                          rows: any,
+                                      ): Promise<void> => {
+                                          const shouldSelectAll =
+                                              this.state.totalNumRows !==
+                                              this.state.numSelectedRows;
+                                          await this.tableRef.current.onAllSelected(
+                                              shouldSelectAll,
+                                          );
+                                          this.setState({
+                                              numSelectedRows: shouldSelectAll
+                                                  ? this.state.totalNumRows
+                                                  : 0,
+                                          });
+                                      },
+                                  },
                                   {
                                       icon: (): JSX.Element => (
                                           <VerifiedIcon data-testid="verify-action" />
                                       ),
                                       tooltip: 'Verify selected rows',
                                       // eslint-disable-next-line @typescript-eslint/no-explicit-any
                                       onClick: (_: any, rows: any): void => {
+                                          if (
+                                              this.state.totalNumRows ===
```

Comment:
this if is repeated 3 times, factor it in a class method please

Reply:
Done.